### ISSUE TITLE
Stepper keyboard navigation does not work with screen reader running

### ISSUE THREAD

**Issue by catherinehynes** [Esri/calcite-components#14707]
**monday.com sync:** #12383019293

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Which Component

Stepper, Stepper Item

### Actual Behavior

Using the stepper with NVDA running does not allow for keyboard interaction except when you hit Ctrl + enter/space or Shift + enter/space. It behaves similarly to the known issue here: https://github.com/Esri/calcite-design-system/issues/11996.

### Expected Behavior

I should be able to use the stepper while a screen reader is on with typical keyboard interactions like enter. 

### Reproduction Sample

https://codepen.io/chynes27/pen/EaZvKYd?editors=100

### Reproduction Steps

1. Open repro [sample](https://codepen.io/chynes27/pen/EaZvKYd?editors=100)
2. Start NVDA
3. Navigate via keyboard to a step (2, 3, or 4) and try to hit enter to activate
4. Note that the enter key doesn't activate the item, and that you can only use the keyboard controls if you hit Ctrl or Shift at the same time

### Reproduction Version

5.1

### Relevant Info

Tested with NVDA on Chrome. I do not see the behavior when using Firefox with NVDA. 

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Enterprise

**Comment by geospatialem** [comment]
### Initial spike/background information
This _may_ be something that relates to AT, browser and web component support limitations similar to https://github.com/Esri/calcite-design-system/issues/11996#issuecomment-2977506468, additional testing and research is needed prior to next steps.

This _may_ also be dependent on some of the markup improvements needed in https://github.com/Esri/calcite-design-system/issues/13537, marking the above as blocked to determine if the role improves the experience.
